### PR TITLE
Create keymap.c

### PR DESCRIPTION
**Pull request by MauliQT**
Edited the OG version with multiple layers and a few macros for Photoshop and "non Working existing Keys" like copy, paste, cut and undo.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
Fixes #
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by MauliQT** [comment]
Just a layout and macro addition

File: keyboards/pabile/p18/keymaps/macro_layer/keymap.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
+ -0,0 +1,77 @@
```
Comment (drashna) [inline keyboards/pabile/p18/keymaps/macro_layer/keymap.c:1]:
```suggestion
```

And could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/pabile/p18/keymaps/macro_layer/keymap.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        if (record->event.pressed) {
+            // when keycode CUT is pressed
+            register_code(KC_LCTL);
+            register_code(X);
+        } else {
+            unregister_code(KC_LCTL)
+            unregister_code(X)
+            // when keycode CUT is released
+        }
+        break;
```
Comment (drashna) [inline keyboards/pabile/p18/keymaps/macro_layer/keymap.c:83]: These can be replaced with `C(KC_C)`, `C(KC_V)`, and `C(KC_X)` respectively, in your keymap.

File: keyboards/pabile/p18/keymaps/macro_layer/keymap.c [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            // when keycode REDO is pressed
+            register_code(KC_LCTL);
+            register_code(Y);
+        } else {
+            unregister_code(KC_LCTL)
+            unregister_code(Y)
+            // when keycode REDO is released
+        }
+        break;        
+    }
```
Comment (drashna) [inline keyboards/pabile/p18/keymaps/macro_layer/keymap.c:117]: Same with these, `C(KC_Z)` amd `C(KC_Y)`.

File: keyboards/pabile/p18/keymaps/macro_layer/keymap.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    return true;
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) { / First encoder below the controller /
```
Comment (drashna) [inline keyboards/pabile/p18/keymaps/macro_layer/keymap.c:122]:
```suggestion
  if (index == 0) { /* First encoder below the controller */
```

File: keyboards/pabile/p18/keymaps/macro_layer/keymap.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      tap_code(KC_VOLD); /volume down/
+    } else {
+      tap_code(KC_VOLU); /volume up/
+                 //Just plain audio controlle
+    }
+  } else if (index == 1) { / Second encoder /
+    if (clockwise) {
+      tap_code16(C(KC_PMNS); /ZOOM IN/               
+    } else {
+      tap_code16(C(KC_PPLS); /ZOOM OUT*/
```
Comment (drashna) [inline keyboards/pabile/p18/keymaps/macro_layer/keymap.c:133]:
```suggestion
      tap_code(KC_VOLD); /* volume down */
    } else {
      tap_code(KC_VOLU); /* volume up */
                 //Just plain audio controlle
    }
  } else if (index == 1) { /* Second encoder */
    if (clockwise) {
      tap_code16(C(KC_PMNS); /* ZOOM IN */               
    } else {
      tap_code16(C(KC_PPLS); /* ZOOM OUT */
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**